src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/FormatterConverter.cs

```diff
@@ -0,0 +1,124 @@
+        }
+
+        [CLSCompliant(false)]
+        public sbyte ToSByte(object value)
+        {
+            if (value == null) ThrowValueNullException();
+            return System.Convert.ToSByte(value, CultureInfo.InvariantCulture);
+        }
+
+        public byte ToByte(object value)
+        {
+            if (value == null) ThrowValueNullException();
+            return System.Convert.ToByte(value, CultureInfo.InvariantCulture);
+        }
+
+        public short ToInt16(object value)
+        {
+            if (value == null) ThrowValueNullException();
+            return System.Convert.ToInt16(value, CultureInfo.InvariantCulture);
+        }
+
+        [CLSCompliant(false)]
+        public ushort ToUInt16(object value)
+        {
+            if (value == null) ThrowValueNullException();
+            return System.Convert.ToUInt16(value, CultureInfo.InvariantCulture);
+        }
+
+        public int ToInt32(object value)
+        {
+            if (value == null) ThrowValueNullException();
+            return System.Convert.ToInt32(value, CultureInfo.InvariantCulture);
+        }
+
+        [CLSCompliant(false)]
+        public uint ToUInt32(object value)
+        {
+            if (value == null) ThrowValueNullException();
+            return System.Convert.ToUInt32(value, CultureInfo.InvariantCulture);
+        }
+
+        public long ToInt64(object value)
+        {
+            if (value == null) ThrowValueNullException();
+            return System.Convert.ToInt64(value, CultureInfo.InvariantCulture);
+        }
+
+        [CLSCompliant(false)]
+        public ulong ToUInt64(object value)
+        {
+            if (value == null) ThrowValueNullException();
+            return System.Convert.ToUInt64(value, CultureInfo.InvariantCulture);
+        }
+
+        public float ToSingle(object value)
+        {
+            if (value == null) ThrowValueNullException();
+            return System.Convert.ToSingle(value, CultureInfo.InvariantCulture);
+        }
+
+        public double ToDouble(object value)
+        {
+            if (value == null) ThrowValueNullException();
+            return System.Convert.ToDouble(value, CultureInfo.InvariantCulture);
+        }
+
+        public decimal ToDecimal(object value)
+        {
+            if (value == null) ThrowValueNullException();
+            return System.Convert.ToDecimal(value, CultureInfo.InvariantCulture);
+        }
+
+        public DateTime ToDateTime(object value)
+        {
+            if (value == null) ThrowValueNullException();
+            return System.Convert.ToDateTime(value, CultureInfo.InvariantCulture);
+        }
+
+        public string ToString(object value)
+        {
+            if (value == null) ThrowValueNullException();
+            return System.Convert.ToString(value, CultureInfo.InvariantCulture);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static void ThrowValueNullException()
```

Comment:
> Do we care that the callstack is different?

I don't think so.  We've done this elsewhere in corefx.

> the NoInlining on the method has me thinking I'm missing something

That was me just being overly cautious about perf.  The JIT currently will inline this throw method but then likely won't inline the caller, whereas with this it's more likely the caller will be inlined.  Not a big deal though either way.